File: Izm.Rumis/Izm.Rumis.Domain/Entities/Resource.cs
using Izm.Rumis.Domain.Attributes;
using Izm.Rumis.Domain.Constants;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Izm.Rumis.Domain.Entities
{
    public class Resource : Entity<Guid>
    {
        [Required]
        [MaxLength(100)]
        public string ResourceNumber { get; set; }

        [MaxLength(500)]
        public string ResourceName { get; set; }

        [MaxLength(100)]
        public string ModelIdentifier { get; set; }

        [Column(TypeName = "decimal(8, 2)")]
        public decimal AcquisitionsValue { get; set; }

        public int ManufactureYear { get; set; }

        [Required]
        [MaxLength(200)]
        public string InventoryNumber { get; set; }

        [Required]
        [MaxLength(200)]
        public string SerialNumber { get; set; }

        public bool? SocialSupportResource { get; set; }

        [Column(TypeName = "longtext")]
        public string ResourceStatusHistory { get; set; }

        [Column(TypeName = "longtext")]
        public string Notes { get; set; }

        public int EducationalInstitutionId { get; set; }
        public virtual EducationalInstitution EducationalInstitution { get; set; }

        [ClassifierType(ClassifierTypes.ResourceSubType)]
        public Guid ResourceSubTypeId { get; set; }
        public virtual Classifier ResourceSubType { get; set; }

        [ClassifierType(ClassifierTypes.ResourceGroup)]
        public Guid? ResourceGroupId { get; set; }
        public virtual Classifier ResourceGroup { get; set; }

        [ClassifierType(ClassifierTypes.ResourceStatus)]
        public Guid ResourceStatusId { get; set; }
        public virtual Classifier ResourceStatus { get; set; }

        [ClassifierType(ClassifierTypes.ResourceLocation)]
        public Guid ResourceLocationId { get; set; }
        public virtual Classifier ResourceLocation { get; set; }

        [ClassifierType(ClassifierTypes.TargetGroup)]
        public Guid TargetGroupId { get; set; }
        public virtual Classifier TargetGroup { get; set; }

        [ClassifierType(ClassifierTypes.ResourceUsingPurpose)]
        public Guid UsagePurposeTypeId { get; set; }
        public virtual Classifier UsagePurposeType { get; set; }

        [ClassifierType(ClassifierTypes.ResourceAcquisitionType)]
        public Guid AcquisitionTypeId { get; set; }
        public virtual Classifier AcquisitionType { get; set; }

        [ClassifierType(ClassifierTypes.ResourceManufacturer)]
        public Guid ManufacturerId { get; set; }
        public virtual Classifier Manufacturer { get; set; }

        [ClassifierType(ClassifierTypes.ResourceModelName)]
        public Guid ModelNameId { get; set; }
        public virtual Classifier ModelName { get; set; }

        public virtual ICollection<ResourceParameter> ResourceParameters { get; set; } = new List<ResourceParameter>();

        public override string ToString()
        {
            if (ModelName == null)
                return $"{Manufacturer.Value}";

            return $"{Manufacturer.Value} {ModelName.Value}";
        }
    }
}

File: Izm.Rumis/Izm.Rumis.App/ClientApp/src/app/types/Resource.ts
export type Resource = {
    id: string
    manufacturer?: {
        id: string
        code: string
        value: string
    },
    inventoryNumber?: string
    serialNumber?: string
    modelName?: {
        id: string
        code: string
        value: string
    }
    educationalInstitution?: {
        id: number
        code: string
        name: string
    }
    resourceName?: string
    modelIdentifier?: string
    resourceNumber?: string
    acquisitionsValue?: number,
    acquisitionType?: {
        id: string
        code: string
        value: string
    }
    manufactureYear?: number
    notes?: string
    resourceGroup?: {
        id: string
        code: string
        value: string
    }
    resourceLocation?: {
        id: string
        code: string
        value: string
    }
    resourceStatus?: {
        id: string
        code: string
        value: string
    }
    resourceStatusHistory?: string
    resourceSubType?: {
        id: string
        code: string
        value: string
    }
    targetGroup?: {
        id: string
        code: string
        value: string
    }
    usagePurposeType?: {
        id: string
        code: string
        value: string
    }
    resourceParameters: {
        id: string;
        value: string;
        parameter: {
            id: string;
            code: string;
            value: string;
        }
    }[]
}

export type ResourceFilterType = {
    resourceNumber?: string;
    inventoryNumber?: string;
    serialNumber?: string;
    resourceName?: string;
    modelNameIds?: string[];
    manufacturer?: string;
    resourceStatusIds?: string[];
    resourceSubTypeIds?: string;
    resourceGroup?: string;
    resourceUsingPurpose?: string;
    targetGroupIds?: string[];
    usagePurposeTypeIds?: string[];
    resourceLocationIds?: string[];
    educationalInstitutionIds?: string;
    supervisorIds?: string[];
    modelIdentifier?: string;
    acquisitionTypeIds?: string;
    sort?: string;
    sortDir?: number;
    page: number;
    take: number;
}